python/test/conftest.py

```diff
@@ -136,3 +138,56 @@ def tempdir(request):
 
     """
     return _create_tempdir(request)
+
+
+@pytest.fixture(scope="function")
+def cg_solver():
+    """Simple Conjugate Gradient solver for SPD problems,
+       which can work in serial or parallel for testing use.
+       Not suitable for large problems."""
+
+    # Basic Conjugate Gradient solver
+    def _cg(comm, A, b, x):
+        kmax = 500
```

Comment:
`maxit` is more intuitive, and should probably be a parameter with a default value.